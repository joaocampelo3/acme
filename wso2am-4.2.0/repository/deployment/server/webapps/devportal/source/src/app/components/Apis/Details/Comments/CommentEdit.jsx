/*
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { FormattedMessage, injectIntl, } from 'react-intl';
import { TextField, Button, Typography } from '@material-ui/core';
import FormControl from '@material-ui/core/FormControl';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import Grid from '@material-ui/core/Grid';
import Alert from '../../../Shared/Alert';
import API from '../../../../data/api';

const styles = theme => ({
    textField: {
        marginTop: 0,
        width: '87.5%',
    },
    contentWrapper: {
        maxWidth: theme.custom.contentAreaWidth,
        paddingLeft: theme.spacing(2),
        paddingTop: theme.spacing.unig,
        marginTop: theme.spacing(2),
    },
    category: {
        width: '12%',
        marginRight: '0.5%',
    },
});

/**
 * Display a component to edit a comment
 * @class CommmentEdit
 * @extends {React.Component}
 */
class CommentEdit extends React.Component {
    /**
     * Creates an instance of CommentEdit
     * @param {*} props properies passed by the parent element
     * @memberof CommentEdit
     */
    constructor(props) {
        super(props);
        this.state = {
            commentText: '',
            category: '',
            currentLength: 0,
        };
        this.inputChange = this.inputChange.bind(this);
        this.handleCategoryChange = this.handleCategoryChange.bind(this);
        this.handleClickUpdateComment = this.handleClickUpdateComment.bind(this);
        this.handleClickCancel = this.handleClickCancel.bind(this);
        this.filterCommentToUpdate = this.filterCommentToUpdate.bind(this);
        this.filterCommentToUpdateReply = this.filterCommentToUpdateReply.bind(this);
    }

    /**
     * @memberof Comments
     */
    componentDidMount() {
        const { comment } = this.props;
        this.setState({
            commentText: comment.commentText,
            category: comment.category,
            currentLength: comment.commentText.length,
        });
    }

    /**
     * Handles the comment text when input changes
     * @param {Object} {target} target element
     * @memberof CommentEdit
     */
    inputChange({ target }) {
        this.setState({ commentText: target.value, currentLength: target.value.length });
    }

    /**
     * Hides the component to edit a comment when cancelled
     * @memberof CommentEdit
     */
    handleClickCancel() {
        const { toggleShowEdit, commentsUpdate, allComments } = this.props;
        toggleShowEdit();
        commentsUpdate(allComments);
    }

    /**
     * Handles category when the category is changed
     * @param {any} event Drop down select event
     * @memberof CommentEdit
     */
    handleCategoryChange(event) {
        this.setState({ category: event.target.value });
    }

    /**
     * Filters the comment to update
     * @memberof CommentAdd
     */
    filterCommentToUpdate(commentToFilter) {
        const { comment } = this.props;
        return commentToFilter.commentId === comment.commentId;
    }

    /**
     * Filters the comment to update
     * @memberof CommentAdd
     */
    filterCommentToUpdateReply(commentToFilter) {
        const { comment } = this.props;
        return commentToFilter.commentId === comment.parentCommentId;
    }

    /**
     * Handles updating a comment
     * @memberof CommentEdit
     */
    handleClickUpdateComment() {
        const {
            apiId, comment, allComments, toggleShowEdit, commentsUpdate, intl,
        } = this.props;
        const { category, commentText } = this.state;
        const Api = new API();
        const commentToEdit = comment;
        commentToEdit.commentText = commentText.trim();
        commentToEdit.category = category;

        // to check whether a string does not contain only whitehis spaces
        if (comment.commentText.replace(/\s/g, '').length) {
            Api.updateComment(apiId, commentToEdit.commentId, commentToEdit)
                .then((result) => {
                    const updatedComment = result.body;
                    if (commentToEdit.parentCommentId === undefined) {
                        const index = allComments.findIndex(this.filterCommentToUpdate);
                        allComments[index].category = updatedComment.category;
                        allComments[index].commentText = updatedComment.commentText;
                    } else {
                        const index = allComments.findIndex(this.filterCommentToUpdateReply);
                        const replyIndex = allComments[index].replies.findIndex(this.filterCommentToUpdate);
                        allComments[index].replies[replyIndex] = updatedComment;
                    }
                    toggleShowEdit();
                    commentsUpdate(allComments);
                })
                .catch((error) => {
                    console.error(error);
                    if (error.response) {
                        Alert.error(error.response.body.message);
                    } else {
                        Alert.error(
                            intl.formatMessage({
                                defaultMessage: 'Something went wrong while adding the comment',
                                id: 'Apis.Details.Comments.CommentEdit.something.went.wrong',
                            }),
                        );
                    }
                });
        } else {
            Alert.error(
                intl.formatMessage({
                    defaultMessage: 'You cannot enter a blank comment',
                    id: 'Apis.Details.Comments.CommentEdit.blank.comment.error',
                }),
            );
        }
    }

    /**
     * Render method of the component
     * @returns {React.Component} Comment html component
     * @memberof CommentEdit
     */
    render() {
        const { classes, theme, intl } = this.props;
        const { category, commentText, currentLength } = this.state;
        return (
            <div>
                <FormControl className={classes.category}>
                    <Select value={category} onChange={this.handleCategoryChange}>
                        <MenuItem value='General'>
                            <FormattedMessage id='Apis.Details.Comments.CommentEdit.general' defaultMessage='General' />
                        </MenuItem>
                        <MenuItem value='Feature Request'>
                            <FormattedMessage
                                id='Apis.Details.Comments.CommentEdit.feature.request'
                                defaultMessage='Feature Request'
                            />
                        </MenuItem>
                        <MenuItem value='Bug Report'>
                            <FormattedMessage
                                id='Apis.Details.Comments.CommentEdit.bug.report'
                                defaultMessage='Bug Report'
                            />
                        </MenuItem>
                    </Select>
                </FormControl>
                <TextField
                    id='multiline-static'
                    autoFocus
                    multiline
                    className={classes.textField}
                    margin='normal'
                    placeholder={intl.formatMessage({
                        defaultMessage: 'Write a comment',
                        id: 'Apis.Details.Comments.CommentEdit.write.a.comment',
                    })}
                    inputProps={{ maxLength: theme.custom.maxCommentLength }}
                    value={commentText}
                    onChange={this.inputChange}
                />
                <Typography className={classes.commentText} align='right'>
                    {currentLength + '/' + theme.custom.maxCommentLength}
                </Typography>
                <Grid container spacing={1}>
                    <Grid item>
                        <Button variant='contained' color='primary' onClick={() => this.handleClickUpdateComment()}>
                            <FormattedMessage id='Apis.Details.Comments.CommentEdit.btn.save' defaultMessage='Save' />
                        </Button>
                    </Grid>
                    <Grid item>
                        <Button onClick={() => this.handleClickCancel()} className={classes.button}>
                            <FormattedMessage
                                id='Apis.Details.Comments.CommentEdit.btn.cancel'
                                defaultMessage='Cancel'
                            />
                        </Button>
                    </Grid>
                </Grid>
            </div>
        );
    }
}

CommentEdit.defaultProps = {
    commentsUpdate: null,
};

CommentEdit.propTypes = {
    classes: PropTypes.instanceOf(Object).isRequired,
    apiId: PropTypes.string.isRequired,
    allComments: PropTypes.instanceOf(Array).isRequired,
    // todo make commentsUpdate required once comment edit feature is supported
    commentsUpdate: PropTypes.func,
    toggleShowEdit: PropTypes.func.isRequired,
    comment: PropTypes.instanceOf(Object).isRequired,
    intl: PropTypes.shape({
        formatMessage: PropTypes.func,
    }).isRequired,
};

export default injectIntl(withStyles(styles, { withTheme: true })(CommentEdit));
