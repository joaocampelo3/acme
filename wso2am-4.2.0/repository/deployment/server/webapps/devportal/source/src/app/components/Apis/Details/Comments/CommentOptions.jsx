/*
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import { FormattedMessage } from 'react-intl';
import AuthManager from '../../../../data/AuthManager';

const styles = theme => ({
    link: {
        color: theme.palette.getContrastText(theme.palette.primary.main),
        '& span.MuiButton-label': {
            color: theme.palette.primary.main,
            fontWeight: '400',
        },
        cursor: 'pointer',
    },
    time: {
        color: theme.palette.getContrastText(theme.palette.background.default),
        marginTop: theme.spacing(0.3),
    },
    verticalSpace: {
        display: 'flex',
        alignItems: 'center',
    },
    disable: {
        color: theme.palette.grey[200],
    },
    commentIcon: {
        color: theme.palette.getContrastText(theme.palette.background.default),
    },
    commentText: {
        color: theme.palette.getContrastText(theme.palette.background.default),
        marginTop: theme.spacing.unig,
        width: '100%',
        whiteSpace: 'pre-wrap',
        overflowWrap: 'break-word',
    },
    root: {
        marginTop: theme.spacing(2.5),
    },
    contentWrapper: {
        maxWidth: theme.custom.contentAreaWidth,
        paddingLeft: theme.spacing(2),
        paddingTop: theme.spacing.unig,
    },
});

/**
 * Component to display options of the comment
 * @class CommentOptions
 * @extends {React.Component}
 */
class CommentOptions extends React.Component {
    /**
     * Creates an instance of CommentAdd
     * @param {*} props properies passed by the parent element
     * @memberof CommentAdd
     */
    constructor(props) {
        super(props);
        this.state = {};
        this.showAddComment = this.showAddComment.bind(this);
        this.showEditComment = this.showEditComment.bind(this);
        this.handleClickOpen = this.handleClickOpen.bind(this);
    }

    /**
     * Shows the component to add a new comment
     * @param {any} index Index of comment in the array
     * @memberof CommentOptions
     */
    showAddComment(replyId) {
        const { showAddComment } = this.props;
        showAddComment(replyId);
    }

    /**
     * Shows the component to edit a comment
     * @param {any} index Index of comment in the array
     * @memberof Comment
     */
    showEditComment(index) {
        const { editIndex, showEditComment } = this.props;
        if (editIndex === -1) {
            showEditComment(index);
        }
    }

    /**
     * Shows the confimation dialog to delete a comment
     * @param {Object} comment Comment that has to be deleted
     * @memberof Comment
     */
    handleClickOpen(comment) {
        const { editIndex, handleClickOpen } = this.props;
        if (editIndex === -1) {
            handleClickOpen(comment);
        }
    }

    /**
     * Returns the date and time in a particular format
     * @param {any} timestamp Timestamp that has to be formatted
     * @memberof CommentOptions
     */
    displayDate(timestamp) {
        const localDate = new Date(timestamp);
        const localDateString = localDate.toLocaleDateString(undefined, {
            day: 'numeric',
            month: 'short',
            year: 'numeric',
        });
        const localTimeString = localDate.toLocaleTimeString(undefined, {
            hour: '2-digit',
            minute: '2-digit',
        });
        return localDateString + ' ' + localTimeString;
    }

    /**
     * Render method of the component
     * @returns {React.Component} Comment html component
     * @memberof CommentOptions
     */
    render() {
        const {
            classes, comment, editIndex, index, theme,
        } = this.props;
        const user = AuthManager.getUser();
        const username = user && user.name;
        const canDelete = (comment.createdBy === username) || user && user.isAdmin();
        return (
            <Grid container className={classes.verticalSpace} key={comment.id}>
                {/* only the comment owner or admin can delete a comment */}
                {canDelete && [
                        <Grid item key='key-delete'>
                            <Button
                                size='small'
                                className={editIndex === -1 ? classes.link : classes.disable}
                                onClick={() => this.handleClickOpen(comment)}
                                color='primary'
                                aria-label={'Delete comment ' + comment.content}
                            >
                                <FormattedMessage
                                    id='Apis.Details.Comments.CommentOptions.delete'
                                    defaultMessage='Delete'
                                />
                            </Button>
                        </Grid>,

                    ]}

                {comment.parentCommentId === null && AuthManager.getUser() && [
                    <Grid item key='key-reply'>
                        <Button
                            size='small'
                            className={classes.link}
                            onClick={() => this.showAddComment(comment.id)}
                            color='primary'
                            aria-label={'Reply to comment ' + comment.content}
                        >
                            <FormattedMessage id='Apis.Details.Comments.CommentOptions.reply' defaultMessage='Reply' />
                        </Button>
                    </Grid>,
                ]}

                {/* only the comment owner can modify the comment from the exact entry point */}
                {/* {comment.createdBy === AuthManager.getUser().name
                    && comment.entryPoint === 'APIStore' && [
                        <Grid item key='key-edit'>
                            <Typography
                                component='a'
                                className={editIndex === -1 ? classes.link : classes.disable}
                                onClick={() => this.showEditComment(index)}
                            >
                                <FormattedMessage
                                    id='Apis.Details.Comments.CommentOptions.edit'
                                    defaultMessage='Edit'
                                />
                            </Typography>
                        </Grid>,
                        <Grid item key='key-edit-verical-divider'>
                            <VerticalDivider height={15} />
                        </Grid>,
                    ]} */}
                {/* <Grid item className={classes.time}>
                    <Typography component='a' variant='caption'>
                        {this.displayDate(comment.createdTime)}
                    </Typography>
                </Grid>
                {/* {editIndex === index
                    ? null
                    : [
                        <Grid item key='key-category-vertical-divider'>
                            <VerticalDivider height={15} />
                        </Grid>,
                        <Grid item className={classes.time} key='key-category'>
                            <Typography component='a' variant='caption'>
                                {comment.category}
                            </Typography>
                        </Grid>,
                    ]}  */}
            </Grid>
        );
    }
}

CommentOptions.defaultProps = {
    showAddComment: null,
};

CommentOptions.propTypes = {
    classes: PropTypes.instanceOf(Object).isRequired,
    editIndex: PropTypes.number.isRequired,
    index: PropTypes.number.isRequired,
    comment: PropTypes.instanceOf(Object).isRequired,
    handleClickOpen: PropTypes.func.isRequired,
    showEditComment: PropTypes.func.isRequired,
    showAddComment: PropTypes.func,
};

export default withStyles(styles, { withTheme: true })(CommentOptions);
