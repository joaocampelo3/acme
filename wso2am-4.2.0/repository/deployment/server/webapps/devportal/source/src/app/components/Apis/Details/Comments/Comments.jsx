/*
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import { Typography } from '@material-ui/core';
import Grid from '@material-ui/core/Grid/Grid';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';
import { FormattedMessage, injectIntl } from 'react-intl';
import classNames from 'classnames';
import CircularProgress from '@material-ui/core/CircularProgress';
import InlineMessage from 'AppComponents/Shared/InlineMessage';
import withSettings from 'AppComponents/Shared/withSettingsContext';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
import Comment from './Comment';
import CommentAdd from './CommentAdd';
import API from '../../../../data/api';
import { ApiContext } from '../ApiContext';
import AuthManager from '../../../../data/AuthManager';

const styles = (theme) => ({
    root: {
        display: 'flex',
        alignItems: 'center',
        paddingTop: theme.spacing(2),
        paddingBottom: theme.spacing(2),
    },
    paper: {
        marginRight: theme.spacing(3),
        paddingBottom: theme.spacing(3),
        paddingRight: theme.spacing(2),
        '& span, & h5, & label, & td, & li, & div, & input': {
            color: theme.palette.getContrastText(theme.palette.background.paper),
        },
    },
    contentWrapper: {
        paddingLeft: theme.spacing(3),
        marginTop: theme.spacing(1),
    },
    contentWrapperOverview: {
        padding: 0,
        width: '100%',
        boxShadow: 'none',
    },
    titleSub: {
        cursor: 'pointer',
        color: theme.palette.getContrastText(theme.palette.background.default),
    },
    link: {
        color: theme.palette.getContrastText(theme.palette.background.default),
        cursor: 'pointer',
    },
    verticalSpace: {
        marginTop: theme.spacing(0.2),
    },
    loadMoreLink: {
        textDecoration: 'none',
    },
    genericMessageWrapper: {
        marginTop: theme.spacing(2),
        marginBottom: theme.spacing(2),
        marginRight: theme.spacing(3),
    },
    paperProgress: {
        padding: theme.spacing(3),
        marginTop: theme.spacing(2),
    },
    button: {
        textTransform: 'capitalize',
    },
});

/**
 * Display a comment list
 * @class Comments
 * @extends {React.Component}
 */
class Comments extends Component {
    static contextType = ApiContext;

    /**
     * Creates an instance of Comments
     * @param {*} props properies passed by the parent element
     * @memberof Comments
     */
    constructor(props) {
        super(props);
        this.state = {
            expanded: true,
            allComments: null,
            comments: [],
            totalComments: 0,
            apiId: null,
            showCommentAdd: false,
        };
        this.handleExpandClick = this.handleExpandClick.bind(this);
        this.handleLoadMoreComments = this.handleLoadMoreComments.bind(this);
        this.toggleCommentAdd = this.toggleCommentAdd.bind(this);
        this.addComment = this.addComment.bind(this);
        this.updateComment = this.updateComment.bind(this);
        this.onDeleteComment = this.onDeleteComment.bind(this);
        this.isCrossTenant = this.isCrossTenant.bind(this);
    }

    /**
     * Gets all the comments for a particular API, when component mounts
     * @memberof Comments
     */
    componentDidMount() {
        let {
            apiId, theme, match, intl, isOverview, setCount,
        } = this.props;
        if (match) apiId = match.params.apiUuid;
        this.setState({ apiId });

        const restApi = new API();
        const limit = theme.custom.commentsLimit;
        const offset = 0;

        restApi
            .getAllComments(apiId, limit, offset)
            .then((result) => {
                let commentList = result.body.list;
                if (isOverview) {
                    setCount(commentList.length);
                    if (commentList.length > 2) {
                        commentList = commentList.slice(commentList.length - 3, commentList.length);
                    }
                }
                this.setState({
                    allComments: commentList,
                    comments: commentList,
                    totalComments: result.body.pagination.total,
                });
            })
            .catch((error) => {
                if (process.env.NODE_ENV !== 'production') {
                    console.log(error);
                }
            });
    }

    /**
     * Handles loading the previous comments
     * @memberof Comments
     */
    handleLoadMoreComments() {
        const { allComments, apiId, comments } = this.state;
        const { theme } = this.props;
        const restApi = new API();
        const limit = theme.custom.commentsLimit;
        const offset = comments.length;

        restApi
            .getAllComments(apiId, limit, offset)
            .then((result) => {
                const newAllCommentList = allComments.concat(result.body.list);
                this.setState({ allComments: newAllCommentList, comments: newAllCommentList });
            })
            .catch((error) => {
                if (process.env.NODE_ENV !== 'production') {
                    console.log(error);
                }
            });
    }

    /**
     * Handles expanding the comment section
     * @memberof Comments
     */
    handleExpandClick() {
        const { expanded } = this.state;
        this.setState({ expanded: !expanded });
    }

    /**
     * Add comment to the comment list
     * @param {any} comment added comment
     * @memberof Comments
     */
    addComment(comment) {
        const { totalComments, allComments } = this.state;
        const { theme: { custom: { commentsLimit } } } = this.props;
        const newTotal = totalComments + 1;

        this.setState({
            allComments: [comment, ...allComments],
            totalComments: newTotal,
            comments: [comment, ...allComments],
        });
    }

    /**
     * Update a specific comment in the comment list
     * @param {any} comment updated comment
     * @memberof Comments
     */
    updateComment(comment) {
        const { comments } = this.state;

        const newComments = comments.reduce((acc, cur) => {
            let temp = cur;
            if (cur.id === comment.id) {
                temp = comment;
            }
            return [...acc, temp];
        }, []);
        this.setState({
            allComments: newComments,
            comments: newComments,
        });
    }

    /**
     * Delete a comment
     * @param {string} commentIdOfCommentToDelete id of deleted commetn
     * @memberof Comments
     */
    onDeleteComment(commentIdOfCommentToDelete) {
        const {
            apiId, comments, totalComments,
        } = this.state;

        const remainingComments = comments.filter((item) => item.id !== commentIdOfCommentToDelete);
        const newTotal = totalComments - 1;

        if (newTotal > remainingComments.length) {
            const restApi = new API();

            restApi
                .getAllComments(apiId, 1, remainingComments.length)
                .then((result) => {
                    if (result.body) {
                        this.setState({
                            totalComments: newTotal,
                            comments: [...remainingComments, ...result.body.list],
                            allComments: [...remainingComments, ...result.body.list],
                        });
                    }
                })
                .catch((error) => {
                    if (process.env.NODE_ENV !== 'production') {
                        console.log(error);
                    }
                });
        } else {
            this.setState({
                totalComments: newTotal,
                comments: remainingComments,
                allComments: remainingComments,
            });
        }
    }

    /**
     * Method to compare the tenant domains
     * @param {*} advertiseInfo advertiseInfo object for the API
     * @param {*} currentUser current logged in user
     * @returns {boolean} true or false
     */
    isCrossTenant(currentUser, tenantDomain) {
        if (!tenantDomain) {
            return false;
        }
        let loggedInUserDomain = null;
        const loggedInUser = currentUser.name;

        if (loggedInUser.includes('@')) {
            const splitLoggedInUser = loggedInUser.split('@');
            loggedInUserDomain = splitLoggedInUser[splitLoggedInUser.length - 1];
        } else {
            loggedInUserDomain = 'carbon.super';
        }

        if (tenantDomain !== loggedInUserDomain) {
            return true;
        } else {
            return false;
        }
    }

    toggleCommentAdd() {
        this.setState((prevState) => ({ showCommentAdd: !prevState.showCommentAdd }));
    }

    /**
     * Render method of the component
     * @returns {React.Component} Comment html component
     * @memberof Comments
     */
    render() {
        const { classes, isOverview, tenantDomain } = this.props;
        const {
            comments, allComments, totalComments, showCommentAdd,
        } = this.state;
        const isCrossTenantUser = AuthManager.getUser() && this.isCrossTenant(AuthManager.getUser(), tenantDomain);
        return (
            <ApiContext.Consumer>
                {({ api }) => (
                    <div
                        className={classNames(
                            { [classes.contentWrapper]: !isOverview },
                            { [classes.contentWrapperOverview]: isOverview },
                        )}
                    >
                        {!isOverview && (
                            <div className={classes.root}>
                                <Typography variant='h4' component='h2' className={classes.titleSub}>
                                    {totalComments + (' ')}
                                    <FormattedMessage id='Apis.Details.Comments.title' defaultMessage='Comments' />
                                </Typography>
                            </div>
                        )}

                        {!isCrossTenantUser && AuthManager.getUser() &&(
                            <Box mt={2} ml={1}>
                                {!showCommentAdd && (
                                    <Button
                                        color='primary'
                                        size='small'
                                        className={classes.button}
                                        startIcon={<AddCircleOutlineIcon />}
                                        onClick={this.toggleCommentAdd}
                                    >
                                        <FormattedMessage
                                            id='Apis.Details.Comments.write.a.new.comment'
                                            defaultMessage='Write a New Comment'
                                        />
                                    </Button>
                                )}
                                {showCommentAdd && (
                                    <CommentAdd
                                        apiId={api.id}
                                        commentsUpdate={this.addComment}
                                        addComment={this.addComment}
                                        allComments={allComments}
                                        replyTo={null}
                                        cancelCallback={this.toggleCommentAdd}
                                        cancelButton
                                    />
                                )}
                            </Box>
                        )}
                        {!allComments && (
                            <Paper className={classes.paperProgress}>
                                <CircularProgress size={24} />
                            </Paper>
                        )}
                        {allComments && totalComments === 0
                            && (
                                <Box mt={2} mb={2} ml={1}>
                                    <InlineMessage
                                        type='info'
                                        title={(
                                            <FormattedMessage
                                                id='Apis.Details.Comments.no.comments'
                                                defaultMessage='No Comments Yet'
                                            />
                                        )}
                                    >
                                        <Typography component='p'>
                                            <FormattedMessage
                                                id='Apis.Details.Comments.no.comments.content'
                                                defaultMessage='No comments available for this API yet'
                                            />
                                        </Typography>
                                    </InlineMessage>
                                </Box>
                            )}
                        <Comment
                            comments={comments}
                            isCrossTenantUser={isCrossTenantUser}
                            apiId={api.id}
                            allComments={allComments}
                            onDeleteComment={this.onDeleteComment}
                            updateComment={this.updateComment}
                        />
                        {totalComments > comments.length && (
                            <div className={classes.contentWrapper}>
                                <Grid container spacing={4} className={classes.root}>
                                    <Grid item>
                                        <Typography className={classes.verticalSpace} variant='body1'>
                                            <a
                                                className={classes.link + ' ' + classes.loadMoreLink}
                                                onClick={this.handleLoadMoreComments}
                                            >
                                                <FormattedMessage
                                                    id='Apis.Details.Comments.load.previous.comments'
                                                    defaultMessage='Show More'
                                                />
                                            </a>
                                        </Typography>
                                    </Grid>
                                    <Grid item>
                                        <Typography className={classes.verticalSpace} variant='body1'>
                                            {'(' + comments.length + ' of ' + totalComments + ')'}
                                        </Typography>
                                    </Grid>
                                </Grid>
                            </div>
                        )}
                    </div>
                )}
            </ApiContext.Consumer>
        );
    }
}

Comments.defaultProps = {
    setCount: () => { },
};
Comments.propTypes = {
    classes: PropTypes.instanceOf(Object).isRequired,
    setCount: PropTypes.func,
};

export default injectIntl(withStyles(styles, { withTheme: true })(withSettings(Comments)));
