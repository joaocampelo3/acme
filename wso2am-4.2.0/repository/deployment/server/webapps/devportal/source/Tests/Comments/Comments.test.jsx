/*
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { unwrap } from '@material-ui/core/test-utils';
import { Typography } from '@material-ui/core';
import Comments from '../../src/app/components/Apis/Details/Comments/Comments';
import Comment from '../../src/app/components/Apis/Details/Comments/Comment';
import CommentAdd from '../../src/app/components/Apis/Details/Comments/CommentAdd';


const CommentsUnwrapped = unwrap(Comments);

let wrapper;
const theme = {
    custom: {
        commentsLimit: 5,
    },
};
const api = {
    id: '6e770272-212b-404e-ab9c-333fdba02f2f',
    isDefaultVersion: true,
    lastUpdatedTime: '2018-11-27T03:08:22.569Z',
    lifeCycleStatus: 'Published',
    name: 'Swagger Petstore',
};

beforeEach(() => {
    wrapper = shallow(<CommentsUnwrapped api={api} apiId={api.id} classes={{}} theme={theme} /> );
});

beforeAll(() => {
    wrapper = shallow(<CommentsUnwrapped api={api} apiId={api.id} classes={{}} theme={theme} /> );
    CommentsUnwrapped.prototype.componentDidMount = () => {
        console.log('componentDidMount method is called');
    };
});


describe('<Comments /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

//     it('should render a <ArrowDropDownCircleOutlined /> with the title of the section - Comments', () => {
//         expect(wrapper.find(ArrowDropDownCircleOutlined)).toHaveLength(1);
//     });

//     it('should render a <Typography /> with the title of the section - Comments', () => {
//         expect(wrapper.find(Typography)).toHaveLength(1);
//     });

//     it('should render a <CommentAdd /> component to add a new comment', () => {
//         expect(wrapper.find(CommentAdd)).toHaveLength(1);
//     });

//     it('should render a <Comment /> component to display a comment', () => {
//         expect(wrapper.find(Comment)).toHaveLength(1);
//     });

//     it('should render a <a /> to load more comments, if the number of comments is greater than the maximum limit to be displayed', () => {
//         wrapper.setState({ startCommentsToDisplay: 1 });
//         expect(wrapper.find('a')).toHaveLength(1);
//     });

//     it('should render 3 <Typography /> s, if the number of comments is greater than the maximum limit to be displayed', () => {
//         wrapper.setState({ startCommentsToDisplay: 1 });
//         expect(wrapper.find(Typography)).toHaveLength(3);
//     });
// });

// describe('<Comments /> interactions', () => {
//     it('should change the state \'expanded\' when the ArrowDropDownCircleOutlined is clicked', () => {
//         const stateBefore = wrapper.state('expanded');
//         wrapper.find(ArrowDropDownCircleOutlined).simulate('click');
//         expect(wrapper.state('expanded')).toEqual(!stateBefore);
//     });

//     it('should change the state \'expanded\' when the Title is clicked', () => {
//         const stateBefore = wrapper.state('expanded');
//         wrapper.find(Typography).first().simulate('click');
//         expect(wrapper.state('expanded')).toEqual(!stateBefore);
//     });

//     it('should not change the state startCommentsToDisplay when the load more comments Arrow is clicked when number of comments is equal to commentsLimit', () => {
//         const stateBefore = wrapper.state('startCommentsToDisplay');
//         wrapper.setState({ startCommentsToDisplay: theme.custom.commentsLimit });
//         wrapper.find(ArrowDropDown).first().simulate('click');
//         expect(wrapper.state('startCommentsToDisplay')).toEqual(stateBefore);
//     });

//     it('should not change the state startCommentsToDisplay when the load more comments Arrow is clicked when number of comments is less than  the commentsLimit', () => {
//         const stateBefore = wrapper.state('startCommentsToDisplay');
//         wrapper.setState({ startCommentsToDisplay: theme.custom.commentsLimit - 1 });
//         wrapper.find(ArrowDropDown).first().simulate('click');
//         expect(wrapper.state('startCommentsToDisplay')).toEqual(stateBefore);
//     });

//     it('should change the state startCommentsToDisplay when the load more comments Arrow is clicked when number of comments is greater than the commentsLimit', () => {
//         const stateBefore = wrapper.state('startCommentsToDisplay');
//         wrapper.setState({ startCommentsToDisplay: theme.custom.commentsLimit + 1 });
//         wrapper.find(ArrowDropDown).first().simulate('click');
//         expect(wrapper.state('startCommentsToDisplay')).toEqual(stateBefore + 1);
//     });

});
