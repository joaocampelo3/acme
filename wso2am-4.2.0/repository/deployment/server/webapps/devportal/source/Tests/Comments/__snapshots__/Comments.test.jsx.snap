// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> rendering renders correctly 1`] = `ShallowWrapper {}`;
